src/test/java/io/reactivex/internal/operators/maybe/MaybeSwitchIfEmptySingleTest.java

```diff
@@ -0,0 +1,108 @@
+/**
+ * Copyright (c) 2016-present, RxJava Contributors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in
+ * compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is
+ * distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See
+ * the License for the specific language governing permissions and limitations under the License.
+ */
+
+package io.reactivex.internal.operators.maybe;
+
+import io.reactivex.Maybe;
+import io.reactivex.Single;
+import io.reactivex.TestHelper;
+import io.reactivex.exceptions.TestException;
+import io.reactivex.functions.Function;
+import io.reactivex.observers.TestObserver;
+import io.reactivex.processors.PublishProcessor;
+import io.reactivex.schedulers.Schedulers;
+import org.junit.Test;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+public class MaybeSwitchIfEmptySingleTest {
+
+    @Test
+    public void nonEmpty() {
+        Maybe.just(1).switchIfEmpty(Single.just(2)).test().assertResult(1);
+    }
+
+    @Test
+    public void empty() {
+        Maybe.<Integer>empty().switchIfEmpty(Single.just(2)).test().assertResult(2);
+    }
+
+    @Test
+    public void error() {
+        Maybe.<Integer>error(new TestException()).switchIfEmpty(Single.just(2))
+        .test().assertFailure(TestException.class);
```

Comment:
Would be great to also check that no values were emitted to check that it didn't switch to fallback Single

Reply:
`assertFailure` does check values:
```
public final U assertFailure(Class<? extends Throwable> error, T... values) {
    return assertSubscribed()
            .assertValues(values)
            .assertError(error)
            .assertNotComplete();
}
```

Reply:
Yes assertFailure does the job under the hood